src/App.js

```diff
@@ -11,19 +11,29 @@ import {
 	ProductListing,
 	NotFound,
 } from "./pages";
-import { Footer, Header } from "./components";
+import { Footer, Header, Loader } from "./components";
+import { useAuth } from "./contexts";
 
 function App() {
 	const currentPath = useLocation().pathname;
```

Comment:
Suggestion, you can restructure **pathname** here. If it's undefined initially, you can add default value to it. 
Ex:
```
const { pathname =' '}=useLocation();
```

Reply:
Updated. Thanks for pointing out.